sycl/include/sycl/ext/oneapi/annotated_arg/annotated_ptr.hpp

```diff
@@ -29,6 +30,46 @@ namespace {
     return *this;                                                              \
   }
 
+// compare strings on compiletime
+template <const char *const &str1, const char *const &str2> struct compareStrs {
+  static constexpr auto impl() noexcept {
+    static_assert(str1 && str2 && "string cannot be nullptr");
+    constexpr const size_t len1 = std::char_traits<char>::length(str1);
+    constexpr const size_t len2 = std::char_traits<char>::length(str2);
+    if (len1 != len2)
+      return false;
+
+    for (int p = 0; p < len1; p++) {
+      if (str1[p] != str2[p])
+        return false;
+    }
+    return true;
+  }
+  static constexpr const bool equal = impl();
+};
+
+// filter properties with AllowedPropsTuple via name checking
+template <typename testProps, typename AllowedPropsTuple>
```

Comment:
Please keep the same style for naming variables, here we have variable1 and Variable2